### ISSUE TITLE
Component Testing for Python SDK

### ISSUE THREAD

**Issue by vigith** [numaproj/numaflow#1794]
# Summary

What change needs making?

# Use Cases

When would you use this?

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by th0ger** [comment]
Yeah we need some better tooling and developer experience on testing.
But it's hard to know what's in the making when only issue titles are filled.
Please include community so we can help you grow. 🙂

**Comment by vigith** [comment]
> Indeed very hard to tell from "test" labels whether is related to numaflow core tests (go/sdk), or test framework for users (pipelines/udf).

We usually tag `area/sdk` when it is SDK related.

**Comment by th0ger** [comment]
I still don't get what this change covers, could you please fill on the issue summary, thanks.
